### PR TITLE
[SCM-979] De-Junit 3

### PR DESCRIPTION
**Pull request by cstamas**
Make all tests junit 4. Please dobule check,
I MAY HAVE MISSED some method named as `testxxx`.

PR targeting other PR https://github.com/apache/maven-scm/pull/142

---

https://issues.apache.org/jira/browse/SCM-979

### CODE REVIEW

**Comment by cstamas** [comment]
Build is ok, but site failed for some reason...

**Comment by cstamas** [comment]
Note: copied over Robert's class PlexusTestCase that is junit4 from maven-resolver for easier "transitioning" of the tests, and next step would be to drop it and use InjectedTest provided by sisu.... but this one single class is not important, it is all the other test classes in project that needs scrutiny.

**Comment by mthmulders** [comment]
I do not think you missed any test methods. Both on master and on your branch, the total tests over all modules is the same:

Run: 59
Failures: 0
Errors: 0
Skipped: 0

File: maven-scm-providers/maven-scm-provider-hg/src/test/java/org/apache/maven/scm/provider/hg/command/blame/HgBlameCommandTckTest.java [hunk 1]
```diff
@@ -26,6 +26,8 @@
 
 import java.util.List;
 
+import static org.junit.Assert.assertEquals;
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-provider-hg/src/test/java/org/apache/maven/scm/provider/hg/command/blame/HgBlameCommandTckTest.java:29]: I don't understand why this import is added; it is the only change in this file?

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/remoteinfo/GitExeRemoteInfoCommandTckTest.java [hunk 2]
```diff
@@ -23,6 +23,8 @@
 import org.apache.maven.scm.provider.git.GitScmTestUtils;
 import org.apache.maven.scm.provider.git.command.remoteinfo.AbstractGitRemoteInfoCommandTckTest;
 
+import static org.junit.Assert.assertEquals;
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/remoteinfo/GitExeRemoteInfoCommandTckTest.java:26]: I don't understand why this import is added; it is the only change in this file?

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommandCommitterAuthorTckTest.java [hunk 3]
```diff
@@ -54,7 +60,7 @@
 {
     
     @Override
-    protected void setUp()
+    public void setUp()
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommandCommitterAuthorTckTest.java:63]: Apparently, this still works (at least, no tests seem to be failing because this doesn't run) but wouldn't it be more nice to annotate with `@Before` (in JUnit 4 style)?

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommandCommitterAuthorTckTest.java [hunk 4]
```diff
@@ -63,7 +69,7 @@ protected void setUp()
     }
     
     @Override
-    protected void tearDown()
+    public void tearDown()
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommandCommitterAuthorTckTest.java:72]: Apparently, this still works (at least, no tests seem to be failing because this doesn't run) but wouldn't it be more nice to annotate with `@After` (in JUnit 4 style)?

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/remoteinfo/JGitRemoteInfoCommandTckTest.java [hunk 5]
```diff
@@ -29,6 +29,8 @@
 import org.apache.maven.scm.provider.git.repository.GitScmProviderRepository;
 import org.eclipse.jgit.util.FileUtils;
 
+import static org.junit.Assert.assertEquals;
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/remoteinfo/JGitRemoteInfoCommandTckTest.java:32]: I don't understand why this import is added; it is the only change in this file?

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/changelog/SvnChangeLogConsumerTest.java [hunk 6]
```diff
@@ -39,19 +40,23 @@
...
-    extends PlexusTestCase
+        extends ScmTestCase
 {
     Logger logger = LoggerFactory.getLogger( getClass() );
 
     SvnChangeLogConsumer consumer;
 
 
-    protected void setUp()
+    public void setUp()
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/changelog/SvnChangeLogConsumerTest.java:59]: Apparently, this still works (at least, no tests seem to be failing because this doesn't run) but wouldn't it be more nice to annotate with `@Before` (in JUnit 4 style)?

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/mkdir/SvnMkdirCommandTest.java [hunk 7]
```diff
@@ -39,7 +42,7 @@
 
     String messageFileString;
 
-    protected void setUp()
+    public void setUp()
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/mkdir/SvnMkdirCommandTest.java:45]: Apparently, this still works (at least, no tests seem to be failing because this doesn't run) but wouldn't it be more nice to annotate with `@Before` (in JUnit 4 style)?

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/mkdir/SvnMkdirCommandTest.java [hunk 8]
```diff
@@ -56,21 +59,23 @@ protected void setUp()
         messageFileString = "--file " + path + " --encoding UTF-8";
     }
 
-    protected void tearDown()
+    public void tearDown()
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/mkdir/SvnMkdirCommandTest.java:62]: Apparently, this still works (at least, no tests seem to be failing because this doesn't run) but wouldn't it be more nice to annotate with `@After` (in JUnit 4 style)?

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/remoteinfo/SvnExeRemoteInfoCommandTckTest.java [hunk 9]
```diff
@@ -24,6 +24,8 @@
 import org.apache.maven.scm.provider.svn.command.remoteinfo.AbstractSvnRemoteInfoCommandTckTest;
 import org.apache.maven.scm.provider.svn.repository.SvnScmProviderRepository;
 
+import static org.junit.Assert.assertTrue;
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/remoteinfo/SvnExeRemoteInfoCommandTckTest.java:27]: I don't understand why this import is added; it is the only change in this file?

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svntest/src/main/java/org/apache/maven/scm/provider/svn/command/blame/SvnBlameCommandTckTest.java [hunk 10]
```diff
@@ -27,6 +27,8 @@
 import java.io.File;
 import java.util.List;
 
+import static org.junit.Assert.assertEquals;
```
Comment (mthmulders) [inline maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svntest/src/main/java/org/apache/maven/scm/provider/svn/command/blame/SvnBlameCommandTckTest.java:30]: I don't understand why this import is added; it is the only change in this file?

File: maven-scm-test/pom.xml [hunk 11]
```diff
@@ -60,7 +60,6 @@
       <groupId>com.google.inject</groupId>
       <artifactId>guice</artifactId>
       <classifier>no_aop</classifier>
-      <scope>runtime</scope>
```
Comment (mthmulders) [inline maven-scm-test/pom.xml:63]: Is this related to dropping JUnit 3?

File: maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java [hunk 12]
```diff
@@ -98,14 +101,14 @@ public abstract void initRepo()
     /**
      * {@inheritDoc}
      */
-    protected void setUp()
+    public void setUp()
```
Comment (mthmulders) [inline maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java:104]: Apparently, this still works (at least, no tests seem to be failing because this doesn't run) but wouldn't it be more nice to annotate with `@Before` (in JUnit 4 style)?

File: maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java [hunk 13]
```diff
@@ -135,10 +138,9 @@ public void removeRepo()
 
     /**
      * Provided to allow removeRepo() to be called. 
-     * @see junit.framework.TestCase#tearDown()
      */
     @Override
-    protected void tearDown()
+    public void tearDown()
```
Comment (mthmulders) [inline maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java:143]: Apparently, this still works (at least, no tests seem to be failing because this doesn't run) but wouldn't it be more nice to annotate with `@After` (in JUnit 4 style)?

File: maven-scm-test/src/main/java/org/apache/maven/scm/ScmTestCase.java [hunk 14]
```diff
@@ -60,7 +63,8 @@
 
     private SecDispatcher secDispatcher;
 
-    protected void setUp()
+    @Override
+    public void setUp()
```
Comment (mthmulders) [inline maven-scm-test/src/main/java/org/apache/maven/scm/ScmTestCase.java:67]: Apparently, this still works (at least, no tests seem to be failing because this doesn't run) but wouldn't it be more nice to annotate with `@Before` (in JUnit 4 style)?

Comment (cstamas) [reply maven-scm-test/src/main/java/org/apache/maven/scm/ScmTestCase.java:69]: fixed

File: maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java (hunk 13 above)
Comment (cstamas) [reply maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java:148]: fixed

File: maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java (hunk 12 above)
Comment (cstamas) [reply maven-scm-test/src/main/java/org/apache/maven/scm/ScmTckTestCase.java:108]: fixed

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/mkdir/SvnMkdirCommandTest.java (hunk 8 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/mkdir/SvnMkdirCommandTest.java:68]: fixed

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/mkdir/SvnMkdirCommandTest.java (hunk 7 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/mkdir/SvnMkdirCommandTest.java:49]: fixed

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/changelog/SvnChangeLogConsumerTest.java (hunk 6 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/changelog/SvnChangeLogConsumerTest.java:61]: fixed

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommandCommitterAuthorTckTest.java (hunk 4 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommandCommitterAuthorTckTest.java:76]: fixed

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommandCommitterAuthorTckTest.java (hunk 3 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommandCommitterAuthorTckTest.java:66]: fixed

File: maven-scm-providers/maven-scm-provider-hg/src/test/java/org/apache/maven/scm/provider/hg/command/blame/HgBlameCommandTckTest.java (hunk 1 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-provider-hg/src/test/java/org/apache/maven/scm/provider/hg/command/blame/HgBlameCommandTckTest.java:29]: In junit3 you inherited this method (w/ same signature from TestCase), but in Junit4 you need to use hamcrest, or Assert or alike, hence static import but no other change (as method sig is same).

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/remoteinfo/GitExeRemoteInfoCommandTckTest.java (hunk 2 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/remoteinfo/GitExeRemoteInfoCommandTckTest.java:26]: In junit3 you inherited this method (w/ same signature from TestCase), but in Junit4 you need to use hamcrest, or Assert or alike, hence static import but no other change (as method sig is same).

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/remoteinfo/JGitRemoteInfoCommandTckTest.java (hunk 5 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/remoteinfo/JGitRemoteInfoCommandTckTest.java:32]: In junit3 you inherited this method (w/ same signature from TestCase), but in Junit4 you need to use hamcrest, or Assert or alike, hence static import but no other change (as method sig is same).

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/remoteinfo/SvnExeRemoteInfoCommandTckTest.java (hunk 9 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/test/java/org/apache/maven/scm/provider/svn/svnexe/command/remoteinfo/SvnExeRemoteInfoCommandTckTest.java:27]: In junit3 you inherited this method (w/ same signature from TestCase), but in Junit4 you need to use hamcrest, or Assert or alike, hence static import but no other change (as method sig is same).

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svntest/src/main/java/org/apache/maven/scm/provider/svn/command/blame/SvnBlameCommandTckTest.java (hunk 10 above)
Comment (cstamas) [reply maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svntest/src/main/java/org/apache/maven/scm/provider/svn/command/blame/SvnBlameCommandTckTest.java:30]: In junit3 you inherited this method (w/ same signature from TestCase), but in Junit4 you need to use hamcrest, or Assert or alike, hence static import but no other change (as method sig is same).

File: maven-scm-test/pom.xml (hunk 11 above)
Comment (cstamas) [reply maven-scm-test/pom.xml:63]: yes, as now Guice classes ARE used in compile scope in maven-scm-test, see Robert's PlexusJunit4TestSupport class, it has import `import com.google.inject.Module;`

Comment (cstamas) [reply maven-scm-test/pom.xml:63]: OTOH, this module is ONLY in test scope of other modules, so guice is compile to this module, while is test + runtime for  others

**Comment by cstamas** [comment]
Found quite some count of missed tests.... so let's see CI results:
Goal: master CI had 1776 tests https://ci-maven.apache.org/job/Maven/job/maven-box/job/maven-scm/job/master/
This PR did ignore ONE test, so 1775 would be good https://ci-maven.apache.org/job/Maven/job/maven-box/job/maven-scm/job/PR-143/

**Comment by cstamas** [comment]
Also, can someone explain why site fails on GH Action? https://github.com/apache/maven-scm/runs/6536237877?check_suite_focus=true

Locally using same commands the build passes.

**Comment by cstamas** [comment]
Also, can someone explain why site fails on GH Action? https://github.com/apache/maven-scm/runs/6536237877?check_suite_focus=true

Locally using same commands the build passes.

**Comment by slawekjaranowski** [comment]
steps to reproduce locally site build error:

```
rm -rf ~/.m2/repository-prv/org/apache/maven/scm/
mvn verify -P run-its
mvn site -P reporting
```

second run - will pass:
```
mvn site -P reporting
```

**Comment by michael-o** [comment]
> steps to reproduce locally site build error:
> 
> ```
> rm -rf ~/.m2/repository-prv/org/apache/maven/scm/
> mvn verify -P run-its
> mvn site -P reporting
> ```
> 
> second run - will pass:
> 
> ```
> mvn site -P reporting
> ```

This I will check, but focusing on tests for now only.

**Comment by michael-o** [comment]
> steps to reproduce locally site build error:
> 
> ```
> rm -rf ~/.m2/repository-prv/org/apache/maven/scm/
> mvn verify -P run-its
> mvn site -P reporting
> ```
> 
> second run - will pass:
> 
> ```
> mvn site -P reporting
> ```

The effect you see is sometimes expected because some reporting plugins require the react with package phase to be run to properly do its work otherwise they will fail. This is such a case: mvn package site -Preporting. Remember that during release an install has already happened.

**Review by michael-o — COMMENTED** [review]
Thank you very much for the work.
